rdkit/Chem/RegistrationHash.py

```diff
@@ -0,0 +1,502 @@
+    stereo_groups = ''
+    if len(smiles_parts) == 2:
+        # note: as with many regex-based things, this is fragile
+        groups = ENHANCED_STEREO_GROUP_REGEX.findall(smiles_parts[1])
+        if groups:
+            # We might have other CXSMILES extensions that aren't stereo groups
+            stereo_groups = f"|{','.join(sorted(groups))}|"
+
+    if stereo_groups:
+        return canonical_smiles + " " + stereo_groups
+
+    return canonical_smiles
+
+
+def GetNoStereoLayers(mol):
+    # SHARED-7909: Strip all Hs, including stereogenic ones.
+    no_stereo_mol = _RemoveUnnecessaryHs(mol)
+    no_stereo_mol.UpdatePropertyCache(False)
+    Chem.rdmolops.RemoveStereochemistry(no_stereo_mol)
+    no_stereo_tautomer_hash = rdMolHash.MolHash(
+        no_stereo_mol, rdMolHash.HashFunction.HetAtomTautomer)
+    no_stereo_smiles = rdMolHash.MolHash(no_stereo_mol,
+                                         rdMolHash.HashFunction.CanonicalSmiles)
+    return (no_stereo_tautomer_hash, no_stereo_smiles)
+
+
+def _GetCanonicalAtomRanksAndBonds(mol, useSmilesOrdering=True):
+    """
+    returns a 2-tuple with:
+
+    1. the canonical ranks of a molecule's atoms
+    2. the bonds expressed as (canonical_atom_rank_1,canonical_atom_rank_2) where
+       canonical_atom_rank_1 < canonical_atom_rank_2
+
+    If useSmilesOrdering is True then the atom indices here correspond to the order of
+    the atoms in the canonical SMILES, otherwise just the canonical atom order is used.
+    useSmilesOrdering=True is a bit slower, but it allows the output to be linked to the
+    canonical SMILES, which can be useful.
+
+    """
+    if mol.GetNumAtoms() == 0:
+        return [], []
+    if not useSmilesOrdering:
+        atRanks = list(Chem.CanonicalRankAtoms(mol))
+    else:
+        smi = Chem.MolToSmiles(mol)
+        ordertxt = mol.GetProp("_smilesAtomOutputOrder")
+        smiOrder = [int(x) for x in ordertxt[1:-1].split(",") if x]
+        atRanks = [0] * len(smiOrder)
+        for i, idx in enumerate(smiOrder):
+            atRanks[idx] = i
+    # get the bonds in canonical form
+    bndOrder = []
+    for bnd in mol.GetBonds():
+        bo = atRanks[bnd.GetBeginAtomIdx()]
+        eo = atRanks[bnd.GetEndAtomIdx()]
+        if bo > eo:
+            bo, eo = eo, bo
+        bndOrder.append((bo, eo))
+    return atRanks, bndOrder
+
+
+def _CanonicalizeDataSGroup(sg,
+                             atRanks,
+                             bndOrder,
+                             fieldNames=None,
+                             sortAtomOrder=True):
+    """
+    NOTES: if sortAtomOrder is true then the atom list will be sorted.
+    This assumes that the order of the atoms in that list is not important
+
+    """
+    if sg.GetProp("TYPE") != "DAT" or not sg.HasProp("FIELDNAME"):
+        return None
+
+    canonicalizableFieldNames = fieldNames or ["Atrop"]
+    fieldName = sg.GetProp("FIELDNAME")
+    if fieldName not in canonicalizableFieldNames:
```

Comment:
I think that is something that we could add later if needed.

If we add it now, we'll never be able to claw it back because we won't know if people rely on it - and neither will they.